packages/flutter_tools/lib/src/http_host_availability_validator.dart

```diff
@@ -0,0 +1,123 @@
+// @dart = 2.9
+import 'base/io.dart';
+import 'base/platform.dart';
+import 'doctor.dart';
+
+/// Hosts used by flutter on all machines
+const List<String> commonRequiredHostUrls = <String>[
+  'https://cloud.google.com/',
+  'https://maven.google.com/',
+  'https://pub.dev/',
+];
+
+/// Hosts used only on MacOS
+const List<String> macOsRequiredHostUrls = <String>[
+  'https://cocoapods.org/',
+];
+
+/// Validation class that checks if all the given URLs are reachable
+class HttpHostAvailabilityValidator extends DoctorValidator {
+  HttpHostAvailabilityValidator({
+    Platform platform,
+    HttpClient httpClient,
+  })   : _platform = platform,
+        _httpClient = httpClient,
+        super('HTTP host availability');
+
+  final Platform _platform;
+  final HttpClient _httpClient;
+
+  @override
+  String get slowWarning => 'HTTP host availability check is taking a long time';
+
+  /// Returns a list of URLs to check availability, different for different platforms
+  List<String> get _allRequiredHosts {
+    if (_platform.isMacOS) {
+      return commonRequiredHostUrls + macOsRequiredHostUrls;
+    } else {
```

Comment:
here and else where, if you have a return in `if/elseif`, you don't need a `else` block, just return instead like:

```
if (a) {
  return c;
}
return d;
```